hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieKeyLocationFetchHandle.java

```diff
@@ -53,7 +54,7 @@ public HoodieKeyLocationFetchHandle(HoodieWriteConfig config, HoodieTable<T, I,
   private List<Pair<HoodieKey, Long>> fetchRecordKeysWithPositions(HoodieBaseFile baseFile) {
     FileFormatUtils fileFormatUtils = HoodieIOFactory.getIOFactory(hoodieTable.getStorage())
         .getFileFormatUtils(baseFile.getStoragePath());
-    if (keyGeneratorOpt.isPresent()) {
+    if (keyGeneratorOpt.isPresent() && baseFile.getStoragePath().getName().contains(HoodieFileFormat.PARQUET.getFileExtension())) {
```

Comment:
I realized that this may cause issue by this, since parquet and orc should use or no use the keygen at the same time. Will fix.

Reply:
Fixed.